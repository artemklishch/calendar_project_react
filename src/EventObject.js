import React from 'react';

const EventObject = ({objectElem}) => {
  return (
    <div className='main__sidebar_day_object' style={{height: '150%'}} >
      <h4>{objectElem.header}</h4>
      <p></p>
    </div>
  );
};
export default EventObject;