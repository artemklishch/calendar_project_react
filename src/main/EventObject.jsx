import React from 'react';
import { forHeightTopAndParagraph } from '../funcs/funcForRenderEvents';

const EventObject = ({ objectElem, onShowFormOnEditing }) => {
  const { top, height, padding, timeString } = forHeightTopAndParagraph(objectElem);
  return (
    <div 
      className='main__sidebar_day_object eventObj'
      style={{ 
        top: top,
        height: height,
        padding: padding,
        backgroundColor: objectElem.backgroundColor }} 
      data-event-id={objectElem._id}
      onClick={onShowFormOnEditing}
    >
      <h4 className="eventObj" data-event-id={objectElem._id}>{objectElem.header}</h4>
      <p className="eventObj" data-event-id={objectElem._id}>{timeString}</p>
    </div>
  );
};
export default EventObject;