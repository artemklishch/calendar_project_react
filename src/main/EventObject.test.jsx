import React from 'react';
import { shallow } from 'enzyme';
import EventObject from './EventObject';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import toJson from 'enzyme-to-json';

describe('EventObject', () => {
  it('should display event', () => {
    const object = {
      header: 'Hello world',
      startDate: new Date(2020,1,1),
      endDate: new Date(2020,2,2),
      _id: 'some-id-1',
      description: 'description',
      backgroundColor: 'red',
    };
    const onShowFormOnEditing = jest.fn();
    const wrappedComponent = shallow(<EventObject objectElem={object} onShowFormOnEditing={onShowFormOnEditing} />);
    expect(toJson(wrappedComponent)).toMatchSnapshot();
  })

  it('should call function for edition', () => {
    const object = {
      header: 'Hello world',
      startDate: new Date(2020,1,1),
      endDate: new Date(2020,2,2),
      _id: 'some-id-1',
      description: 'description',
      backgroundColor: 'red',
    };
    const onShowFormOnEditing = jest.fn();
    const wrappedComponent = shallow(<EventObject objectElem={object} onShowFormOnEditing={onShowFormOnEditing} />);
    wrappedComponent.find('.main__sidebar_day_object').simulate('click');
    expect(onShowFormOnEditing).toBeCalled();
  })
});