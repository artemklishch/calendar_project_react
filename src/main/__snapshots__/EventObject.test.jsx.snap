// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventObject should display event 1`] = `
<div
  className="main__sidebar_day_object eventObj"
  data-event-id="some-id-1"
  onClick={[MockFunction]}
  style={
    Object {
      "backgroundColor": "red",
      "height": "70560%",
      "padding": undefined,
      "top": undefined,
    }
  }
>
  <h4
    className="eventObj"
    data-event-id="some-id-1"
  >
    Hello world
  </h4>
  <p
    className="eventObj"
    data-event-id="some-id-1"
  >
    00:00 - 00:00
  </p>
</div>
`;
